### PR TITLE
NAS-119933 / 23.10 / Enhance validation for licensed systems in apps

### PR DESCRIPTION
**Pull request by sonicaj**
This commit adds changes preventing licensed systems to create more catalogs or update their preferred trains to anything other then enterprise train.

### CODE REVIEW

**Comment by Ornias1993** [comment]
This is pretty-much a complete and utter dick-head move towards other projects wanting to release enterprise-level Apps.

This should be "the default" for licensed systems that want to use Apps, with a license flag to override (with thorough warnings that systems will not get support form iX for App issues if the license flag is set).

At this point, it basically invalidates **ALL** the work we, at @truecharts have been putting into preparing for SMB/Enterprise clients/users. It's simply not-done and brings up the question why I didn't even get a headsup when LITERALLY asking morgan about Enterprise train plans a 1-2 months ago.

**Comment by yocalebo** [comment]
Do you know what this check actually does?

**Comment by Ornias1993** [comment]
> Do you know what this check actually does?

Wait, deleted my comment… because the code shows it DOES offer an “enabled” flag. >.<

**Comment by yocalebo** [comment]
Right, so all this does is prevent our customers that have purchased officially supported hardware from changing the train on the chart. Idea is to keep them from running 3rd party apps since we are contractually obligated to supporting them.

**Comment by Ornias1993** [comment]
> Right, so all this does is prevent our customers that have purchased officially supported hardware from changing the train on the chart. Idea is to keep them from running 3rd party apps since we are contractually obligated to supporting them.

Okey, so it DOES prevent third party catalogs after all?

**Comment by yocalebo** [comment]
Why do you care about that? Are you expecting to offer paid support for truechart apps to our enterprise customers?

**Comment by Ornias1993** [comment]
> Why do you care about that? Are you expecting to offer paid support for truechart apps to our enterprise customers?

TLDR: yes, our at least to our enterprise train. Lets assume the target audience has an overlap for now. details are still a fair bit TBD, when it comes to scope and pricing though…

An idea originally brought up by Morgen and (re)discussed only 1-3 months ago.

**Review by Ornias1993 — COMMENTED** [review]
I've thoroughly checked the code by now.
This PR should not cause needless issues, even for TrueCharts.

- The 'train' limitation only required trains for enterprise users to be explicitly called "enterprise", which is a good way to ensure train creators explicitly need to define enterprise users.
-  The JAILS license flag, enables all catelogs regardless of origin


Note:
In the future an additional flag might need to be added to seperate "JAILS" (AKA: DIY applications and external catalogs) from the iX Official Enterprise Apps.

But after latest update by @sonicaj I see no issues having this merged anymore.
LGTM/Approved.

**Review by yocalebo — APPROVED** [review]
